src/main/java/uk/gov/hmcts/darts/audio/component/impl/AudioRequestBeingProcessedFromArchiveQueryImpl.java

```diff
@@ -73,4 +76,24 @@ public List<AudioRequestBeingProcessedFromArchiveQueryResult> getResults(Integer
         );
     }
 
+    @Override
+    public List<AudioRequestBeingProcessedFromArchiveQueryResult> getResultsByMediaIds(List<Integer> mediaIds) {
+        return jdbcTemplate.query(
+            AUDIO_REQUEST_QUERY +
```

Comment:
I don't think you can use the same base query for yours as it's possible that no audio requests has happened yet for that hearing-media.
We have the `hearing_id` that can be used for the WHERE clause.
Then I think the FROM can start from `hearing` rather than `media_request` table.


**GET /audio/hearings/{hearing_id}/audios**

I think you probably want a separate `MediaBeingProcessedFromArchiveQuery` then you wouldn't need to worry about the media_request stuff.

Looking at postgres_app_v62_sec_v20_ret_v7 – Data Dictionary
(https://github.com/hmcts/darts-api/blob/dmp-2156-preview-audio-unavailable/src/main/resources/db/reference/modernised_darts_erd.pdf)
```
Table: darts.media
Linked from Table: darts.hearing_media_ae
Links to Table: darts.hearing
```

Reply:
modified the query etc to be based on hearing id - we now have a separate AudioBeingProcessedFromArchiveQuery - note that both this and the original query both return a list of AudioRequestBeingProcessedFromArchiveQueryResult - have left this named as it is rather than complicating things by renaming to something even more ghastly like AudioRequestorAudioBeingProcessedFromArchiveQueryResult

Reply:
Yep, I don't think it should be renamed to something like that.
I introduced it originally to represent the result model for my query around the notification: **audio_request_being_processed_from_archive**

It acts on the **Audio Requests** POST endpoint (media_request context) whereas yours is an **Audio** GET endpoint - for determining if preview should be available (media context)